core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestServer.scala

```diff
@@ -139,7 +139,9 @@ private[rest] class StandaloneSubmitRequestServlet(
     val driverExtraLibraryPath = sparkProperties.get("spark.driver.extraLibraryPath")
     val superviseDriver = sparkProperties.get("spark.driver.supervise")
     val appArgs = request.appArgs
-    val environmentVariables = request.environmentVariables
+    // Filter SPARK_LOCAL environment variables from being set on the remote system.
+    val environmentVariables =
+      request.environmentVariables.filterNot(_._1.startsWith("SPARK_LOCAL"))
```

Comment:
Could you please check the code to ensure that `SPARK_LOCAL_DIRS` has not been used as a global config?